rp2040-hal/src/i2c.rs

```diff
@@ -241,18 +241,17 @@ macro_rules! hal {
             impl<Sda: PinId + BankPinId, Scl: PinId + BankPinId>
                 I2C<$I2CX, (Pin<Sda, FunctionI2C>, Pin<Scl, FunctionI2C>)> {
                 /// Configures the I2C peripheral to work in master mode
-                pub fn $i2cX<F, SystemF>(
+                pub fn $i2cX<F>(
                     i2c: $I2CX,
                     sda_pin: Pin<Sda, FunctionI2C>,
                     scl_pin: Pin<Scl, FunctionI2C>,
                     freq: F,
                     resets: &mut RESETS,
-                    system_clock: SystemF) -> Self
+                    system_clock: super::clocks::PeripheralClock) -> Self
```

Comment:
I don't think ownership of `PeripheralClock` can be passed to the I2C controller: That would fail as soon as you try to initialize a second peripheral which also needs to now the current clock frequency.

Perhaps pass a reference, and to avoid lifetime issues, don't store the reference in the struct, but extract the actual frequency in the constructor, and only store that value?
(BTW: The parameter probably shouldn't be called `system_clock`. I guess that's the reason why you made the mistake of passing the system clock frequency.)

Reply:
> extract the actual frequency in the constructor, and only store that value?

We already don't ever use it after the constructor. If we did need it again, we'd need to look it up.
Maybe that's a better answer, but would need the infra to do that lookup - I'm keen to do it but wanted to get other parts done first.
But for now, taking a ref should be good, i'll do that.

> The parameter probably shouldn't be called system_clock

I'll need to check if it's supposed to take the system clock or peripheral clock, though I suspect you are correct.

Reply:
Going to move this back to draft until I can work out lifetimes :(